modules/http/src/main/java/org/mule/module/http/internal/listener/grizzly/ResponseStreamingCompletionHandler.java

```diff
@@ -103,6 +104,10 @@ public void completed(WriteResult result)
                 {
                     doComplete();
                 }
+                else if (isDone && httpResponsePacket.getRequest().getMethod().equals(HEAD))
```

Comment:
Can we analyse whether we should force an `EmptyHttpEntity` at some other point instead?